std/bitmanip.d

```diff
@@ -886,6 +886,130 @@ public:
         return b;
     }
 
+    /**
+      Sets all the values in the $(D BitArray) to the
+      value specified by $(D val).
+     */
+    void opSliceAssign(bool val)
+    {
+        _ptr[0 .. fullWords] = val ? ~size_t(0) : 0;
+        if (endBits)
+            _ptr[fullWords] = val ?
+                _ptr[fullWords] | endMask :
+                _ptr[fullWords] & ~endMask;
+    }
+
+    ///
+    @system unittest
+    {
+        import std.algorithm.comparison : equal;
+
+        auto b = BitArray([1, 0, 1, 0, 1, 1]);
+
+        b[] = true;
+        // all bits are set
+        assert(b.bitsSet.equal([0, 1, 2, 3, 4, 5]));
+
+        b[] = false;
+        // none of the bits are set
+        assert(b.bitsSet.empty);
+    }
+
+    /**
+      Sets the bits of a slice of $(D BitArray) which
+      starts at the index $(D start) and ends at the
+      index ($D end) - 1 with the values specified by $(D val).
+     */
+    void opSliceAssign(bool val, size_t start, size_t end)
+    in
+    {
+        assert(start >= 0 && start <= _len);
+        assert(end >=0 && end <= _len);
```

Comment:
That is not correct. `length` is a property which simply returns `_len`. `dim` is the property which returns the number of words. I though I'd use _len to avoid the extra function call.